src/NuGet.Clients/NuGet.VisualStudio.Common/VsHierarchyUtility.cs

```diff
@@ -56,6 +61,31 @@ public static string[] GetProjectTypeGuids(EnvDTE.Project project)
             return projectTypeGuids;
         }
 
+        public static bool IsSupported(IVsHierarchy hierarchy, string projectTypeGuid)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (hierarchy.IsCapabilityMatch("AssemblyReferences + DeclaredSourceItems + UserSourceItems"))
```

Comment:
Similar to `EnvDTEProjectUtility.IsProjectCapabilityCompliant`. Consider unifying them.